consensus/src/block_storage/sync_manager.rs

```diff
@@ -64,11 +64,11 @@ impl BlockStore {
     /// Check if we're far away from this ledger info and need to sync.
     /// This ensures that the block referred by the ledger info is not in buffer manager.
     pub fn need_sync_for_ledger_info(&self, li: &LedgerInfoWithSignatures) -> bool {
-        // TODO move min gap to fallback (30) to config, and if configurable make sure the value is
+        // TODO move min gap to fallback to config, and if configurable make sure the value is
         // larger than buffer manager MAX_BACKLOG (20)
         (self.ordered_root().round() < li.commit_info().round()
             && !self.block_exists(li.commit_info().id()))
-            || self.commit_root().round() + 30.max(2 * self.vote_back_pressure_limit)
+            || self.commit_root().round() + 1000.max(2 * self.vote_back_pressure_limit)
```

Comment:
What's the idea here? by 1000 rounds, blocks would be GCd, so only way to sync is via state sync?

Reply:
this condition is used to help when validators can follow consensus but not execution/commit e.g. execution being too slow. the race condition was when we decide to sync and the time we start sync the execution/commit catches up. so increase the number to mitigate the race, and yeah good point if data is GCd, they can only wait to sync, maybe I should use 100 or something